content/blog/2018-02-07-update-on-async-rendering.md

```diff
@@ -0,0 +1,173 @@
+---
+title: Update on Async Rendering
+author: [bvaughn, gaearon]
+---
+
+For the past few months, the React team has been experimenting with [asynchronous rendering](/blog/2017/09/26/react-v16.0.html#new-core-architecture), and we are very excited about the new features it enables.
+
+Along the way, our research has shown that some of our component lifecycles tend to encourage unsafe coding practices. They are:
+
+* `componentWillMount`
+* `componentWillReceiveProps`
+* `componentWillUpdate`
+
+These lifecycle methods have often been misunderstood and subtly misused; furthermore, we anticipate that these potential problems may be more prominent with async rendering. Because of this, we are adding an "UNSAFE_" prefix to these lifecycles in a future release. React [follows semantic versioning](/blog/2016/02/19/new-versioning-scheme.html), so the migration path will be gradual:
+
+* **16.3**: Introduce aliases for the unsafe lifecycles, `UNSAFE_componentWillMount`, `UNSAFE_componentWillReceiveProps`, and `UNSAFE_componentWillUpdate`. (Both the old lifecycle names and the new aliases will work in this release.)
+* **16.4**: Enable deprecation warning for `componentWillMount`, `componentWillReceiveProps`, and `componentWillUpdate`. (Both the old lifecycle names and the new aliases will work in this release.)
+* **17.0**: Remove `componentWillMount`, `componentWillReceiveProps`, and `componentWillUpdate` . (Only the new "UNSAFE_" lifecycle names will work in this release.)
+
```

Comment:
Any plans to have a codemod to auto-prefix with `UNSAFE`?

Reply:
Yes, there is [`rename-unsafe-lifecycles`](https://github.com/reactjs/react-codemod/blob/master/transforms/rename-unsafe-lifecycles.js), but it is intended for updating application code and so will probably not be mentioned until the deprecation warnings have been turned on with 16.4.

The 16.3 release is intended to give library authors a head start on fixing async things in advance of the deprecation warnings being turned on. Unfortunately, just renaming the methods won't actually fix any of the potential async problems, so I don't think mentioning the codemod is a good idea for this release.